### DISCUSSION TITLE
It is recommended to delete settings.json in the vSCode directory

### DISCUSSION THREAD

**Discussion by davidgxj** [arendst/Tasmota#17997] [Q&A]
### PROBLEM DESCRIPTION
_It is recommended to delete settings.json in the vSCode directory, because it compiles errors in the vSCode under Windows_

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [X] Tasmota binary firmware version number used: _12.4.0,master,development____
  - [ ] Pre-compiled
  - [X] Self-compiled

Platformio-IDE.TOOLBAR of Platformio IDE (V3.0.0) plugin is a good feature. Under the new Tasmota, there is a new Toolbar custom expansion. Because I used VSCODE under Windows to compile, it turned out that it caused it to not pass ENV parameters during compilation.

For example:
When you want to compile Env: Tasmota32, click PLATFORMIO: Build, CMD should be 
```lua
platformio.exe run --environment tasmota32
```
Now is 
```lua
platformio.exe run
```
Lack of parameters.

**Comment by Jason2866** [comment]
There is no passing of any env parameter. Have you updated platformio to latest version?

**Reply by davidgxj** [reply]
PlatformIO core 6.1.6
vscode PlatformIO IDE 3.0.0
vscode 1.75.1
windwos default vscode settings.json
`    "platformio-ide.toolbar": [
        {
            "text": "$(home)",
            "tooltip": "PlatformIO: Home",
            "commands": "platformio-ide.showHome"
        },
        {
            "text": "$(check)",
            "tooltip": "PlatformIO: Build",
            "commands": "platformio-ide.build"
        },
        {
            "text": "$(arrow-right)",
            "tooltip": "PlatformIO: Upload",
            "commands": "platformio-ide.upload"
        },
        {
            "text": "$(trashcan)",
            "tooltip": "PlatformIO: Clean",
            "commands": "platformio-ide.clean"
        },
        {
            "text": "$(beaker)",
            "tooltip": "PlatformIO: Test",
            "commands": "platformio-ide.test"
        },
        {
            "text": "$(plug)",
            "tooltip": "PlatformIO: Serial Monitor",
            "commands": "platformio-ide.serialMonitor"
        },
        {
            "text": "$(terminal)",
            "tooltip": "PlatformIO: New Terminal",
            "commands": "platformio-ide.newTerminal"
        }
    ]
`

**Comment by Jason2866** [comment]
Just tested with Windows too, does work as intended. The commands for compile and upload etc. are the same as used in standard Platformio bottom menu bar. Only new and additional is the compile, erase and upload icon.
Sometimes Platformio has a hickup when settings changes. Delete all platformio hidden and cache folders.

**Comment by davidgxj** [comment]
I just tried it, the new vScode, Platformio environment test, found the same, screenshot.
![image](https://user-images.githubusercontent.com/17197318/219968104-0e1039e0-7aec-4a45-9568-ca513987a46a.png)

**Reply by barbudor** [reply]
Work as expected
Have you upgraded to Platformio 3.0.0 ?
In case of doubt, 
- Quit VSCode
- Erase completely:
  - `%HOME%\.platformio`
  - `Tasmota\.pio`
  - `Tasmota\.cache`
- Relaunch VSCode and let the PlatformIO extension re-install PIO from scratch

![image](https://user-images.githubusercontent.com/2996042/219968874-13d7696d-6c3d-476c-be0a-004da5ea43c6.png)

**Reply by davidgxj** [reply]
@barbudor ，Your settings.json is not running. After running, icon should be

![image](https://user-images.githubusercontent.com/17197318/219970184-705c4bd6-6b51-4dbb-9633-d40e77e4efb4.png)


My configuration

![image](https://user-images.githubusercontent.com/17197318/219970286-f3c761dc-4b71-479a-9dd5-e3cfc41f73a1.png)

**Reply by barbudor** [reply]
And how is that supposed to happen ?
VScode should read it automatically

**Reply by barbudor** [reply]
@davidgxj 
I have reproduced with Windows 11
What Windows are you using ?

**Comment by Jason2866** [comment]
Thx @barbudor for extensive tests with Windows11. Issue arrives when "workspaces" from VSC is used.

**Reply by barbudor** [reply]
When workspace is used, the settings.json is not loaded
What I reproduced is without workspaces, just openning the folder

**Comment by Jason2866** [comment]
The problem is a platformio VSC IDF issue when switching env. Quick fix removing custom Tasmota toolbar :-(
https://github.com/platformio/platformio-vscode-ide/issues/2623
https://github.com/platformio/platformio-vscode-ide/issues/1697#issuecomment-1420289631

**Comment by davidgxj** [comment]
Thx @barbudor @Jason2866 

I saw that the problem in the latest git was fixed.

Today I used some time to test this issue.

result:

ubuntu 20.04
windows 11
windows 10

This bug has all.
